### ISSUE TITLE
RouteNotFoundException on untrusted domain access

### ISSUE THREAD

**Issue by patschi** [nextcloud/server#5347]
### Steps to reproduce
1. Just waiting until anyone tries to access nextcloud using untrusted domain

### Expected behaviour
No error.

### Actual behaviour
Following error showed up:
```
Level: Error
App: no app in context
Message:
Symfony\Component\Routing\Exception\RouteNotFoundException: Unable to generate a URL for the named route "theming.Theming.getLogo" as such route does not exist.
/var/www/nextcloud/lib/private/Route/Router.php - line 339: Symfony\Component\Routing\Generator\UrlGenerator->generate('theming.Theming...', Array, 1)
/var/www/nextcloud/lib/private/Route/CachingRouter.php - line 60: OC\Route\Router->generate('theming.Theming...', Array, false)
/var/www/nextcloud/lib/private/URLGenerator.php - line 70: OC\Route\CachingRouter->generate('theming.Theming...', Array)
/var/www/nextcloud/apps/theming/lib/ThemingDefaults.php - line 156: OC\URLGenerator->linkToRoute('theming.Theming...')
/var/www/nextcloud/lib/public/Defaults.php - line 186: OCA\Theming\ThemingDefaults->getLogo(true)
/var/www/nextcloud/core/templates/layout.guest.php - line 45: OCP\Defaults->getLogo()
/var/www/nextcloud/lib/private/Template/Base.php - line 176: include('/var/www/nextcl...')
/var/www/nextcloud/lib/private/Template/Base.php - line 151: OC\Template\Base->load('/var/www/nextcl...', NULL)
/var/www/nextcloud/lib/private/legacy/template.php - line 202: OC\Template\Base->fetchPage(NULL)
/var/www/nextcloud/lib/private/legacy/template.php - line 224: OC_Template->fetchPage()
/var/www/nextcloud/lib/private/Template/Base.php - line 132: OC_Template->fetchPage()
/var/www/nextcloud/lib/base.php - line 809: OC\Template\Base->printPage()
/var/www/nextcloud/lib/base.php - line 1082: OC init()
/var/www/nextcloud/index.php - line 38: require_once('/var/www/nextcl...')
{main}

Level: Warning
App: core
Message:
Trusted domain error. "<any_client_ip>" tried to access using "<cloud_ip>" as host.
```

### Server configuration

**Operating system**: Ubuntu 16.04 LTS
**Web server:** nginx
**Database:** MariaDB
**PHP version:** 7.0
**Nextcloud version:** 12.0.0.29
**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh

**Comment by patschi** [comment]
@blizzz Should this issue being considered as `starter issue` and `feature: theming`?

**Comment by blizzz** [comment]
I don't know whether it's a starter one, but theming related it is. 

cc @nextcloud/theming @LukasReschke

**Comment by MorrisJobke** [comment]
I can reproduce:

* set a logo in the theming app
* remove the domain you are using from the trusted_domains
* open the Nextcloud

**Comment by MorrisJobke** [comment]
Fix is in #5821
